### PR TITLE
Add GVAR_FUNC macro and adapt code to use it

### PR DESCRIPTION
**Pull request by fingolfin**
This is an alternative to PR #1546

### CODE REVIEW

**Comment by markuspf** [comment]
Apart from the test failures from changed error messages, I'd like to think that this PR is the chance to give the other tables the same treatment? (`GVarFilts`, `GVarOpers`, and friends)

File: src/gap.c [hunk 1]
```diff
@@ -2814,128 +2814,51 @@ void ThreadedInterpreter(void *funcargs) {
...
+    GVAR_FUNC(HANDLE_OBJ, 1, "object"),
+    GVAR_FUNC(LoadedModules, 0, ""),
+    GVAR_FUNC(WindowCmd, 1, "arg-list"),
+    GVAR_FUNC(MicroSleep, 1, "msecs"),
+    GVAR_FUNC(Sleep, 1, "secs"),
+    GVAR_FUNC(GAP_EXIT_CODE, 1, "exit code"),
+    GVAR_FUNC(QUIT_GAP, -1, "args"),
+    GVAR_FUNC(FORCE_QUIT_GAP, -1, "args"),
     { "SHOULD_QUIT_ON_BREAK", 0, "",
       FuncShouldQuitOnBreak, "src/gap.c:FuncShouldQuitOnBreak"},
```
Comment (fingolfin) [inline src/gap.c]: For examples like this, I'd consider renaming the C function to match the GAP one. OK?

File: src/gvars.c [hunk 2]
```diff
@@ -1481,39 +1481,20 @@ static StructGVarFunc GVarFuncs [] = {
     { "MakeReadWriteGVar", 1, "name",
       MakeReadWriteGVarHandler, "src/gap.c:MakeReadWriteGVar" },
```
Comment (fingolfin) [inline src/gvars.c]: Quite some entries use the suffix `Handler` instead of the prefix `Func`. I'll ignore them for now.

**Comment by fingolfin** [comment]
@markuspf sure we could add macros for those other tables, too. For filters/attributes/properties, a single additional macro would suffice. For opers, we'd need a third one. But there are very few of those anyway.

**Comment by fingolfin** [comment]
I just pushed a version that fixes one regression; but (at least) one error / diff in the test suite remains, for out-of-tree-builds
```
# Input is:
Display(RETURN_FIRST);
# Expected output:
function ( object... )
    <<kernel code from src/gap.c:RETURN_FIRST>>
end
# But found:
function ( object... )
    <<kernel code from ../../src/gap.c:RETURN_FIRST>>
end
```
Personally, I consider the second output a plus, i.e. for me the relative path is more convenient. But it makes the test fail, and I am not quite how to fix it. 

But of course the issue would go away if we only stored / printed the `src/gap.c` bit here. I could modify the kernel to omit the `../../`, in various ways. Question is: should I?

**Comment by stevelinton** [comment]
It's not obvious to most users what the base directory of these paths is (typically the out-of-tree build directory) so basing them at a GAP root would seem quite sensible if it's easy. Also from a testing perspective, if the relative path from the build directory to the source affects output it will be next to impossible to make a test file that works reliably.

**Comment by fingolfin** [comment]
@stevelinton added a commit to resolve this. Let's see if the tests pass.

**Pull request merged**